snprintf

santana opened the pull request:
sprintf is inherently unsafe, easy to misuse. snprintf makes more robust code, it is harder to misuse.

Commit:
makes squeak.cog.spur compile on OpenBSD 6.0

Commit:
adds OpenBSD support to Squeak's shell script wrapper

Commit:
replaces most sprintf with snprintf

Commit:
replaces sprintf with snprintf

Commit:
replaces bash with sh

eliotmiranda committed:
Add missing new lines at end of Pharo mac makefiles

eliotmiranda committed:
Add pointer to cmake distribution for pharo VM builds on Mac.

eliotmiranda committed:
Add a reminder to put cmake in PATH.

bencoman commented on platforms/unix/vm/sqUnixExternalPrims.c:
```diff
@@ -452,11 +441,7 @@ ioFindExternalFunctionIn(char *lookupName, void *moduleHandle)
 	signed char *accessorDepthVarPtr;
 
-# ifdef HAVE_SNPRINTF
 	snprintf(buf+strlen(buf), sizeof(buf), "AccessorDepth");
```
btw, shouldn't this existing code be...
  snprintf(buf+strlen(buf), sizeof(buf) - strlen(buf), "AccessorDepth");

santana replied:
wow, yes,  you have just spotted a current bug :)

bencoman commented on platforms/unix/vm/sqUnixExternalPrims.c:
```diff
@@ -202,11 +203,7 @@ void *ioLoadModule(char *pluginName)
 
   /* try dlopen()ing LIBRARY_PREFIX<name>LIBRARY_SUFFIX searching only the default locations modulo LD_LIBRARY_PATH et al */
 
-# if defined(HAVE_SNPRINTF)
```
hi Gerardo,  
A good initiative, but these HAVE_SNPRINTF checks indicate some platforms don't support snprintf().  However this code is from 2010 and maybe there is broader support now. But in particular MSVC didn't support snprintf() until 2015. I do see Nicolas reports successful builds using MSVC2015 - so I guess that would become a the minimum requirement for native Windows.  

Any platforms missing snprintf() might pick up a portable version [1] from instead #defining snprintf() as sprintf()...
[1] https://github.com/search?utf8=%E2%9C%93&q=snprintf

santana replied:
I agree with you on using a portable snprintf where it is not available. That means two things:
1. Those HAVE_SNPRINTF would go away anyways, because instead of being used everywhere in the source code, it will be used in a single place: the Makefiles, for conditionally linking the portable version.
2. I think that platforms we care about support snprintf, so there are no reasons for working around in the code for the lack of snprintf (and even more so when, as you pointed out, we can use a portable version)

Commit:
bug fix: too many characters were allowed to be written to buf.

eliotmiranda committed:
CogVM source as per VMMaker.oscog-eem.1958

Fix VMProfileMacSupportPlugin for 64-bits.

eliotmiranda committed:
Remove wring-headed 32-bit pcs in 64-bit VMs support because both Mac OS X and
Linux use the full 64-bit address space for code.

codefrau committed:
Remove MIDIPlugin from builds (but leave #80 open to track its rewrite)

eliotmiranda committed:
CogVM source as per VMMaker.oscog-eem.1959

Fix primitiveConstantFill for Spur 16-bit and 64-bit indexable objects.
Ensure that on Spur bytes beyond the last byte are zero.

Fix compilation of the Cogits by exporting full block entry offsets correctly.

eliotmiranda committed:
VMMaker.oscog-eem.1960

Fix off-by-one error in the offset of the byte containing
the "is full block method" flag in a CogBlockMethod.
